Tests/SwiftDiagnosticsTest/GroupDiagnosticsFormatterTests.swift

```diff
@@ -60,6 +60,31 @@ extension GroupedDiagnostics {
 }
 
 final class GroupedDiagnosticsFormatterTests: XCTestCase {
+  func testSourceLocations() {
```

Comment:
It'd be nice to put the actual file's name somewhere as well, but I have no great suggestions as to where to put it.

Reply:
With `#sourceLocation` we have two "files" - the real file on disk (usually generated in this case) and the "presumed" file (the one noted in `#sourceLocation`). After this change, the file noted in the diagnostic is the presumed file, which matches the Swift compiler. That makes sense since `#sourceLocation` was added for a reason, but I have found that knowing the underlying file is useful too.

That would have been more clear if I put this comment on the actual `#sourceLocation` line :)